File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/view/adapters/SearchSetAdapter.java
package com.dbottillo.mtgsearchfree.view.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.dbottillo.mtgsearchfree.R;
import com.dbottillo.mtgsearchfree.model.MTGSet;

import java.util.List;

public class SearchSetAdapter extends ArrayAdapter<MTGSet> {

    public SearchSetAdapter(Context context, List<MTGSet> sets) {
        super(context, -1, sets);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final SetHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.row_spinner_item, parent, false);
            holder = new SetHolder(convertView);
            convertView.setTag(holder);
            convertView.setId(position);
        } else {
            holder = (SetHolder) convertView.getTag();
        }
        holder.name.setText(getItem(position).getName());
        return convertView;
    }

    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        final SetHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.row_spinner_dropdown_item, parent, false);
            holder = new SetHolder(convertView);
            convertView.setTag(holder);
            convertView.setId(position);
        } else {
            holder = (SetHolder) convertView.getTag();
        }
        holder.name.setText(getItem(position).getName());
        return convertView;
    }

    static class SetHolder {
        TextView name;

        SetHolder(View row) {
            name = (TextView) row.findViewById(android.R.id.text1);
        }
    }

}

File: feature_search/src/main/kotlin/com/dbottillo/mtgsearchfree/search/SearchSetAdapter.kt
package com.dbottillo.mtgsearchfree.search

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.dbottillo.mtgsearchfree.model.MTGSet

class SearchSetAdapter(context: Context, sets: List<MTGSet>) : ArrayAdapter<MTGSet>(context, -1, sets) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = convertView
                ?: LayoutInflater.from(context).inflate(R.layout.row_spinner_item, parent, false)
        val holder: SetHolder
        if (convertView == null) {
            holder = SetHolder(view)
            view.tag = SetHolder(view)
            view.id = position
        } else {
            holder = convertView.tag as SetHolder
        }
        holder.name.text = getItem(position).name
        return view
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view = convertView
                ?: LayoutInflater.from(context).inflate(R.layout.row_spinner_dropdown_item, parent, false)
        val holder: SetHolder
        if (convertView == null) {
            holder = SetHolder(view)
            view.tag = holder
            view.id = position
        } else {
            holder = convertView.tag as SetHolder
        }
        holder.name.text = getItem(position).name
        return view
    }

    class SetHolder(row: View) {
        var name = row.findViewById<TextView>(android.R.id.text1)
    }
}
